### PR TITLE
First take at Thinking in Flutter, with styles and images.

### PR DESCRIPTION
**Pull request by LarkAscending**
**HIGH IMPACT** This change replaces the splash screen on flutter.io.

Welcome to Thinking in Flutter Lite! (big shoutout @abarth). 

Please review for scope and audience appropriateness. Feel free to note any content that seems granular, superfluous, or opaque. If you find the tone too playful, please comment (keeping in mind we want to invite people in).

## Notes
* Images are provisional and not final. If you'd like tweaks or more/different images, feedback is welcome.
* The images override a couple of site-wide CSS properties: max-width for the widget class hierarchy and float for the layer cake (to wrap text around it).
* The layer cake is an included HTML file, not an image file. This is hopefully temporary...
    * It has its own embedded styles.
    * The tooltips are commented out...they won't function on mobile so will probably be removed. 
    * Overriding CSS pseudoclasses is tricky. The no_icon class in _sass/_customstyles had to change to remove the blank line where the outbound icon used to be. While THIS CSS CHANGE AFFECTS THE WHOLE SITE, today no other pages use the [nifty no_icon class](https://github.com/flutter/website/search?utf8=%E2%9C%93&q=no_icon). It enables the links in the layer cake to just be links, without the FontAwesome icon that comes after links in text.
*  Markdown links are inline. There are many, making the reference-at-the-end thing seem like the wrong approach. If mandatory :( let me know. 

Everything has been tested locally. Staged version to come...

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
- - [Bootstrap into Dart](/bootstrap-into-dart/) if you are new to the language
- for developing Flutter apps.
+![Shrine, iOS left and Android right](/images/whatisflutter/shrine_both.png)
 
-## Developer Guides
+This is an online marketplace app from the [Gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery/lib/demo), 
+a showcase of Flutter apps to download and run on a phone. Shrine has 
+high-quality scrolling images, interactive cards, buttons and dropdown lists, 
+and a shopping cart page. Run Shrine on an iPhone or Android phone; the UI 
+appearance and behavior are equivalent. Feel free to browse through the source
```
Comment (abarth) [inline index.md]: equivalent -> Do we want to emphasize that point?  Folks might want their apps to adapt more to the target platform.  We might just want to say that you can run it on both platforms.

**Comment by abarth** [comment]
* widget_class_hierarchy.png.  Rather than calling out ProxyWidget explicit, I'd replace with `...` to indicate that there are more kinds of exotic widgets.  As drawn, it's raises too much of a question about "what is proxy widget"?

* state.png.  I'd have the line to the state come out of the right side of the PopupMenuButton circle to indicate that the relationship between PopupMenuButton and State is different than the relationship between StatefulWidget and PopupMenuButton.

File: index.md [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-title: Flutter is a mobile app SDK for iOS and Android
+title: Welcome to Flutter
```
Comment (sethladd) [inline index.md]: Can you rebase to master? This was changed recently.

File: index.md [hunk 3]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
+mobile apps for Android and iOS. The goal is to deliver apps that feel natural 
+on different platforms, and we embrace differences like scrolling behavior, 
+overscroll behaviors, and more. 
 
- - [Set up your machine](/setup/) with Flutter and iOS/Android tools.
- - [Get started](getting-started) to run your first app on Android and iOS.
- - [A tour of the Flutter framework](widgets-intro)
- - [Bootstrap into Dart](/bootstrap-into-dart/) if you are new to the language
- for developing Flutter apps.
+![Shrine, iOS left and Android right](/images/whatisflutter/shrine_both.png)
```
Comment (sethladd) [inline index.md]: We should make this demo remove the shadows in the iOS version. That would make for a much stronger example.

File: index.md (hunk 1 above)
Comment (sethladd) [inline index.md]: Maybe: "similar, but customized to look and feel natural on each platform"

File: index.md [hunk 4]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
-## Support
+*   Be highly productive
+    *   Develop for iOS and Android from a single codebase
+    *   Do more with less code, even on a single OS, with a modern, expressive 
+language and a declarative approach
+*   Prototype and iterate quickly
+    *   Experiment by changing code and reloading as your app runs
+    *   Fix crashes and continue debugging from where the app left off
+*   Create beautiful, highly-customized user experiences
+    *   Leverage a complete set of self-referential Material Design widgets
```
Comment (sethladd) [inline index.md]: what does "self-referential" mean?

File: index.md [hunk 5]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
+    *   Realize whatever visual design you have in mind, without the limitations
```
Comment (sethladd) [inline index.md]: I don't know if we can state this fully. For example, we don't do 3D.

File: index.md (hunk 1 above)
Comment (LarkAscending) [inline index.md]: Yep, I wondered about that...

How about "Run Shrine on an iPhone or Android phone, then check
out the single codebase [here](https://github.com/flutter/flutter/tree/master/examples) 
in our github repository."

File: index.md [hunk 6]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
+widget previously returned. The framework does the heavily lifting of comparing 
+the previous build with the current build and determining what modifications 
+need to be made to the user interface. 
+
+This automated comparison is surprisingly effective, enabling high-performance,
+interactive apps. And the design of the build function simplifies your code by
+focusing on declaring what a widget is made of, rather than the complexities of
+updating the user interface from one state to another.
+
+### Handling User Interaction
```
Comment (sethladd) [inline index.md]: do we capitalize each word in our titles? Or do we use sentence-case ?

File: _includes/layer-cake.html [hunk 7]
```diff
@@ -0,0 +1,145 @@
+<html>
```
Comment (sethladd) [inline _includes/layer-cake.html]: I don't think we need this tag here. We're already including this file in an html file.

**Comment by abarth** [comment]
sounds good

File: _includes/layer-cake.html [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+.tooltip:hover .tooltiptext {
+    visibility: visible;
+    opacity: 1;
+}
+
+</style>
+
+<body>
```
Comment (sethladd) [inline _includes/layer-cake.html]: I don't think we need this tag here.

File: _includes/layer-cake.html [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+<tr>
+  <td class="the-abyss" align="center" colspan="8" background-color="#1976D2">
+      <div>&hellip;
+<!--        <span class="tooltiptext">Deeper layers</span> -->
+      </div>
+  </td>
+</tr>
+</table>
+
+</body>
```
Comment (sethladd) [inline _includes/layer-cake.html]: we can remove these two tags

File: _includes/layer-cake.html [hunk 10]
```diff
@@ -0,0 +1,145 @@
...
+    <a class="simplelink no_icon" href="https://docs.flutter.io/flutter/material/material-library.html">
+      <div >Material
+ <!--       <span class="tooltiptext">Widgets implementing Material Design</span> -->
+      </div>
+    </a>
+  </td>
+  <td class="high-level" align="center" width="50%">
+    <a class="simplelink no_icon" href="https://docs.flutter.io/flutter/cupertino/cupertino-library.html">
+      <div>Cupertino
+<!--        <span class="tooltiptext">iOS widgets set</span> -->
```
Comment (sethladd) [inline _includes/layer-cake.html]: could we remove the comments?

File: index.md [hunk 11]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
+## Why use Flutter?
 
- - [Widgets overview](widgets)
- - [Gestures in Flutter](gestures)
- - [Animations in Flutter](animations)
- - [Using custom fonts](custom-fonts)
- - [Layered design (video)](https://www.youtube.com/watch?v=dkyY9WCGMi0)
- - [Rendering pipeline (video)](https://www.youtube.com/watch?v=UUfXWzp0-DU)
- - [Dealing with box constraints in Flutter](layout)
+What are some advantages of Flutter? It helps you:
```
Comment (sethladd) [inline index.md]: We might also want a bullet on technical advantages:

* world-class text rendering
* fast startup and performance
* industry standard 2D->GPU library

File: index.md [hunk 12]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
-
- - [Source on GitHub](https://github.com/flutter/flutter)
- - [Architecture diagram](https://docs.google.com/presentation/d/1cw7A4HbvM_Abv320rVgPVGiUP2msVs7tfGbkgdrTy0I/edit?usp=sharing)
- - [Issue tracker](https://github.com/flutter/flutter/issues)
- - [Writing useful bug reports](bug-reports)
- - [FAQ](faq)
- - [Design principles](design-principles)
-
-## Support
+*   Be highly productive
```
Comment (sethladd) [inline index.md]: Could we mention Hot Reload here?

File: index.md [hunk 13]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
+    *   Develop for iOS and Android from a single codebase
+    *   Do more with less code, even on a single OS, with a modern, expressive 
+language and a declarative approach
+*   Prototype and iterate quickly
```
Comment (sethladd) [inline index.md]: Is this the same as "highly productive" ?

File: index.md (hunk 5 above)
Comment (sethladd) [inline index.md]: Maybe: "Build customized, beautiful, brand-driven designs"

File: index.md [hunk 14]
```diff
@@ -9,55 +9,192 @@ iOS and Android from a single codebase.
...
+[Padding](https://docs.flutter.io/flutter/widgets/Padding-class.html),
+[DecoratedBox](https://docs.flutter.io/flutter/widgets/DecoratedBox-class.html),
+and [Transform](https://docs.flutter.io/flutter/widgets/Transform-class.html)
+widgets.  Rather than subclassing Container to produce a customized effect, you
+can compose these, and other, simple widgets in novel ways.
+
+The class hierarchy is shallow and broad to maximize the possible number of 
+combinations.
+
+![Widget class hierarchy](/images/whatisflutter/Widget_class_hierarchy.png){:style="max-width: 900px"}
```
Comment (sethladd) [inline index.md]: Can we use the Roboto font in these images?

File: index.md (hunk 2 above)
Comment (LarkAscending) [inline index.md]: Harsher words were never spoken... 
:(

File: index.md (hunk 14 above)
Comment (LarkAscending) [inline index.md]: Someone else may be able to, but I don't have the font locally. Let's make sure the images are right and the ones we want first...

File: index.md (hunk 2 above)
Comment (sethladd) [inline index.md]: technically don't have to rebase. Manually switching this back is sufficient :)

File: index.md (hunk 4 above)
Comment (LarkAscending) [inline index.md]: The idea is, they're made with our own framework. They prove the concept and we ate our own dogfood.
We started with "self-hosted". "Self-referential" means they demonstrate what they're made of. Agree that we need a less-fancy word for this... ideas?

File: _includes/layer-cake.html (hunk 10 above)
Comment (LarkAscending) [inline _includes/layer-cake.html]: Do we have the analytics on device types? If most folks are on mobile devices and we don't have to recreate them, they can be removed. If we have a lot of desktop users and we might recreate them, easier to leave them.

File: index.md (hunk 4 above)
Comment (LarkAscending) [inline index.md]: changed to "self-describing"

File: index.md (hunk 13 above)
Comment (LarkAscending) [inline index.md]: Well, the "highly productive" bullet is really strategic stuff. But the impact of the word "strategic" is nil on a developer audience.

Indented "Prototype" and its sub-bullets under "Productive" for now.

File: index.md (hunk 11 above)
Comment (LarkAscending) [inline index.md]: Prefer to keep it simple here...however, there may be other pages where we can work in this info.

File: index.md (hunk 12 above)
Comment (LarkAscending) [inline index.md]: Mentioned it in a lower bullet. It is a cool feature...but we're working hard to keep this page high-level.

File: _includes/layer-cake.html (hunk 10 above)
Comment (sethladd) [inline _includes/layer-cake.html]: curious, what function do these commented tag provide?

File: index.md (hunk 4 above)
Comment (sethladd) [inline index.md]: We're also not complete (and the spec continues to evolve so not sure we'll every be "complete"). One idea:

"Use a near-complete set of Material design widgets, built with the Flutter framework."

I believe we lower-case design across the site, when used with "Material".

near-complete is a little more humble, and more accurate.

"built with the Flutter framework" is perhaps more clear.

File: index.md (hunk 12 above)
Comment (sethladd) [inline index.md]: Fair enough.

File: index.md (hunk 11 above)
Comment (sethladd) [inline index.md]: For the homepage, I strongly encourage us to enumerate the technical advantages of the product. Somewhere... not sure where on the page :)

File: index.md (hunk 14 above)
Comment (sethladd) [inline index.md]: where are the sources for these images?

You can install Roboto from here: https://fonts.google.com/specimen/Roboto . (select "Select this font" and then you get a download link)

File: index.md [hunk 15]
```diff
@@ -9,55 +9,191 @@ iOS and Android from a single codebase.
...
-## Support
+*   Be highly productive
+    *   Develop for iOS and Android from a single codebase
+    *   Do more with less code, even on a single OS, with a modern, expressive 
+language and a declarative approach
+    *   Prototype and iterate easily
+        *   Experiment by changing code and reloading as your app runs (with hot reload)
+        *   Fix crashes and continue debugging from where the app left off
+*   Create beautiful, highly-customized user experiences
+    *   Leverage a complete set of self-describing Material Design widgets
```
Comment (abarth) [inline index.md]: We should find a word other than "leverage".  That word has legal connotations that we don't intend.  Also, the Material Design widgets are not self-describing.

Comment (sethladd) [inline index.md]: Weird, I thought I left a comment on this. Let me try again :)

The Material spec is evolving, so it's safer to say "near-complete".

Agree with Adam, the word "leverage" is also overloaded and may not be clear to our users what we mean.

Maybe: "a near-complete set of Material widgets, built on the Flutter framework".  This does two things: A) shows that our framework can produce a professional widget set, and B) differentiates us from other frameworks that drive OEM widgets.

File: index.md (hunk 3 above)
Comment (LarkAscending) [inline index.md]: Is this a must-have for the PR?

File: index.md (hunk 15 above)
Comment (LarkAscending) [inline index.md]: Changed to "Reap the benefits of"... 

Can we call the Material widget set "house-made"? ;-)

Comment (LarkAscending) [inline index.md]: I called them a "rich" set of widgets, since 'robust' has become cliche'd and with "near-complete", it sounds kind of unfinished. And honestly, they're cool and great.

Comment (sethladd) [inline index.md]: "rich" works: "Use a rich set of material widgets, built with the Flutter framework."

Has a nice ring to it :)

File: _includes/layer-cake.html (hunk 10 above)
Comment (LarkAscending) [inline _includes/layer-cake.html]: They save work later when someone doesn't have to recreate the tooltip text. They're correct. If it's important, LMK. I'll add it to the contributing doc, not to leave comments in.

File: index.md (hunk 2 above)
Comment (sethladd) [inline index.md]: Can we change the title back to the more descriptive title?

File: index.md (hunk 3 above)
Comment (sethladd) [inline index.md]: We won't block this PR. We have some work to do to make a more iOS-y version of this screenshot.

File: index.md (hunk 15 above)
Comment (sethladd) [inline index.md]: "Rich" works. "Use a rich set of material widgets, built with Flutter's framework" has a nice ring to it :)

File: index.md (hunk 2 above)
Comment (LarkAscending) [inline index.md]: Oh, I thought this was a stop-gap change, waiting for the real page to come.
Tell you what...after this PR if people who come to the website are still confused about what Flutter is, we can definitely change it back.

File: index.md (hunk 11 above)
Comment (LarkAscending) [inline index.md]: Will definitely keep an eye out for ways to work in these selling points, and justify them so that users don't question whether we can fairly make the claims. This was recent UX feedback we got - on "high-performance". It was hard work to string together a stripped-down narrative to show what we meant, why it was fact-based and not just marketing.

Flutter is great, and it does these things, and let's work at demonstrating that.

File: index.md (hunk 15 above)
Comment (LarkAscending) [inline index.md]: Done!

File: index.md (hunk 14 above)
Comment (LarkAscending) [inline index.md]: Looks like the fonts in KolourPaint (open source Linux drawing tool from kde.org) are ones that are included with the tool. There is nothing in the UI for adding fonts, no folder location, no documentation on adding fonts, no Google search results. Installing Roboto using your steps has no effect on what can be selected in the UI. I think this tool is limited in terms of fonts.

There is no "source" format - these are PNG images that can be edited.

File: _includes/layer-cake.html (hunk 10 above)
Comment (LarkAscending) [inline _includes/layer-cake.html]: Could you please add me so I can see the analytics for flutter.io? I'm used to using metrics to inform doc decisions...they've become something I rely on for the occasional reality check.

File: index.md (hunk 3 above)
Comment (LarkAscending) [inline index.md]: OK, will track that and replace as soon as possible.

File: index.md (hunk 2 above)
Comment (sethladd) [inline index.md]: Updating the title isn't so much about clarifying for people visiting the homepage, it's for SEO. When we update the title, it gives search results a much more clear one-liner. Seeing a one-liner for what Flutter is, is much more helpful than "Welcome to Flutter" in Google search results.

File: index.md (hunk 14 above)
Comment (LarkAscending) [inline index.md]: I think the way forward here is to use a loaner Mac, install Omnigraffle and font, then re-draw the images. Should be ready to go in 1 week but it might slip out post-First Look.

File: index.md (hunk 2 above)
Comment (LarkAscending) [inline index.md]: Ah! Have we considered getting software to help with that? Was looking at [this gem](https://github.com/jekyll/jekyll-seo-tag) for help with SEO on another site.

LMK how you'd like to proceed. I believe it's the kind of thing you can only test in a limited way locally, i.e. that it doesn't break stuff... More research needed.

Comment (sethladd) [inline index.md]: I think for now, we can update this title back to "Mobile app SDK for iOS and Android".

And yeah, would love to add even more metadata! Good idea.

Comment (LarkAscending) [inline index.md]: Good - feel free to do that in a separate PR...I'm not comfortable as it's not standard industry practice in documentation, to let the SEO tail wag doc titles. A technical writer prioritizes usability and recognition for new users.

Comment (sethladd) [inline index.md]: Usability might be inclusive of "how do people find our docs via search". But no problem, happy to do another PR.

File: _includes/layer-cake.html [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+
+}
+
+
+</style>
+
+<table cellpadding="0">
+<tr>
+  <td class="high-level optional" align="center" width="50%">
+    <a class="simplelink no_icon" href="https://docs.flutter.io/flutter/material/material-library.html">
```
Comment (abarth) [inline _includes/layer-cake.html]: Why does custom link to the material-library?

File: index.md [hunk 17]
```diff
@@ -1,63 +1,200 @@
...
 {% include note.html content="_Flutter is an early-stage open-source project_. We hope you try Flutter out and send us [feedback](mailto:flutter-dev@googlegroups.com)." %}
 
-## New to Flutter
+## What is Flutter?
 
-See these topics to quickly start building cross-platform mobile apps with Flutter.
+Flutter provides an easy and productive way to build and deploy high-performance 
+mobile apps for Android and iOS. The goal is to deliver apps that feel natural 
+on different platforms, and we embrace differences like scrolling behavior, 
+overscroll behaviors, and more.
```
Comment (abarth) [inline index.md]: It seems weak to mention scrolling twice.  Maybe typography is a better second example?

File: index.md [hunk 18]
```diff
@@ -1,63 +1,200 @@
...
- - [A tour of the Flutter framework](widgets-intro)
- - [Bootstrap into Dart](/bootstrap-into-dart/) if you are new to the language
- for developing Flutter apps.
+![Shrine, iOS left and Android right](/images/whatisflutter/shrine_both.png)
 
-## Developer Guides
+This is an online marketplace app from the [Gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery/lib/demo), 
+a showcase of Flutter apps to download and run on a phone. Shrine has 
+high-quality scrolling images, interactive cards, buttons and dropdown lists, 
+and a shopping cart page. Run Shrine on an iPhone or Android phone, then check
```
Comment (abarth) [inline index.md]: > interactive cards, buttons and dropdown lists, and a shopping cart page

I don't quite follow the grammar here.  Specifically I wonder if the first "and" should be a "," instead.

Comment (abarth) [inline index.md]: > on an iPhone or Android phone

We'd normally say "on iOS or Android".

File: index.md [hunk 19]
```diff
@@ -1,63 +1,200 @@
...
+out the single codebase for this and more examples [here](https://github.com/flutter/flutter/tree/master/examples) 
+in our github repository.
```
Comment (abarth) [inline index.md]: github is spelled GitHub.

File: index.md [hunk 20]
```diff
@@ -1,63 +1,200 @@
...
+
+No mobile development experience is required to get started. Apps are written 
+in [Dart](https://dartlang.org), which will look familiar if you've used a
```
Comment (abarth) [inline index.md]: Missing trailing `/` in URL.

File: index.md [hunk 21]
```diff
@@ -1,63 +1,200 @@
...
-## Support
+*   Be highly productive
+    *   Develop for iOS and Android from a single codebase
+    *   Do more with less code, even on a single OS, with a modern, expressive 
+language and a declarative approach
+    *   Prototype and iterate easily
+        *   Experiment by changing code and reloading as your app runs (with hot reload)
+        *   Fix crashes and continue debugging from where the app left off
+*   Create beautiful, highly-customized user experiences
+    *   Benefit from a rich set of Material design widgets built using
```
Comment (abarth) [inline index.md]: The D in Design is capitalized for Material Design.  See https://material.io/

File: index.md [hunk 22]
```diff
@@ -1,63 +1,200 @@
...
+        Flutter's own framework
+    *   Realize custom, beautiful, brand-driven designs, without the limitations 
+of OEM widget sets
+
+## Core principles
+
+Flutter includes a functional-reactive framework, a 2D rendering engine, ready-
+made widgets, and development tools. The components work together to help you
```
Comment (abarth) [inline index.md]: Will ready-made show up on the site without a space after the - even though there's a line break there?

Comment (abarth) [inline index.md]: > The components

These components ?

File: index.md [hunk 23]
```diff
@@ -1,63 +1,200 @@
...
+
+Flutter includes a functional-reactive framework, a 2D rendering engine, ready-
+made widgets, and development tools. The components work together to help you 
+design, build, test, and debug apps. Everything is organized around a few core 
+principles. 
+
+### Everything's a Widget
+
+Widgets are the basic building blocks of every Flutter app. Each widget is an 
+immutable declaration of part of a user interface.  Unlike other frameworks that
```
Comment (abarth) [inline index.md]: > part of a user interface

part of the user interface

File: index.md [hunk 24]
```diff
@@ -1,63 +1,200 @@
...
+*   a structural element (like a button or menu)
+*   a stylistic element (like a font or color scheme)
+*   an aspect of layout (like padding)
+*   some business logic 
+*   and so on...
+
+Widgets form a hierarchy based on composition.  Each widget nests inside, and 
+inherits properties from, its parent.  There is no separate "application" object.
+Instead, the root widget serves this role.
+
```
Comment (abarth) [inline index.md]: extra blank line here

File: index.md [hunk 25]
```diff
@@ -1,63 +1,200 @@
...
+![Widget class hierarchy](/images/whatisflutter/Widget_class_hierarchy.png){:style="max-width: 900px"}
+
+You can also control the *layout* of a widget by composing it with other widgets.
+For example, to center a widget, you wrap it in a Center widget. There are
+widgets for padding, alignment, row, columns, and grids. These layout widgets
+do not show themselves in an app's UI.  Instead, their sole purpose is to
+control some aspect of another widget's layout. To understand why a widget
+renders in a certain way, it's often helpful to inspect the neighboring widgets.
+
+#### Layer cakes are fun and functional
```
Comment (abarth) [inline index.md]: boo!

File: index.md [hunk 26]
```diff
@@ -1,63 +1,200 @@
...
+
+The Flutter framework is organized into a series of layers, with each layer
+building upon the previous layer.
+
+{% include layer-cake.html %}
+
+The diagram shows the upper layers of the framework, which are used more 
+frequently than the lower layers. For the complete set of libraries, see 
+Flutter's [API documentation](https://docs.flutter.io).
```
Comment (abarth) [inline index.md]: Missing trailing `/` in URL

File: index.md [hunk 27]
```diff
@@ -1,63 +1,200 @@
...
+Nothing is hidden from you.  You reap the productivity benefits of a high-level, 
+unified widget concept, without sacrificing the ability to dive as deeply as you 
+wish into the lower layers.
+
+### Building widgets
+
+You define the unique characteristics of a widget by implementing a 
+[build](https://docs.flutter.io/flutter/widgets/StatelessWidget/build.html) 
+function that returns a tree of widgets. This tree represents the widget's part 
+of the user interface in concrete terms.  For example, a toolbar widget might
```
Comment (abarth) [inline index.md]: I'd say "in more concrete" terms.  They might not yet be fully concrete.

File: index.md [hunk 28]
```diff
@@ -1,63 +1,200 @@
...
+
+These widgets subclass [StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html)
+(rather than [StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html))
+and store their mutable state in a subclass of [State](https://docs.flutter.io/flutter/widgets/State-class.html).
+
+![Separate state and widget objects](/images/whatisflutter/state.png)
+
+Whenever you mutate a State object (e.g., increment the counter), you must call
+[setState](https://docs.flutter.io/flutter/widgets/State/setState.html)() to 
+signal the framework to update the user interface by calling the State build
```
Comment (abarth) [inline index.md]: .. State's build method ...

File: index.md [hunk 29]
```diff
@@ -1,63 +1,200 @@
...
+These widgets subclass [StatefulWidget](https://docs.flutter.io/flutter/widgets/StatefulWidget-class.html)
+(rather than [StatelessWidget](https://docs.flutter.io/flutter/widgets/StatelessWidget-class.html))
+and store their mutable state in a subclass of [State](https://docs.flutter.io/flutter/widgets/State-class.html).
+
+![Separate state and widget objects](/images/whatisflutter/state.png)
+
+Whenever you mutate a State object (e.g., increment the counter), you must call
+[setState](https://docs.flutter.io/flutter/widgets/State/setState.html)() to 
+signal the framework to update the user interface by calling the State build 
+method again. For an example of managing state, see the [MyApp template](https://github.com/flutter/flutter/blob/fa2caa737bdde4d72525d9d4260abb0ba5548c01/packages/flutter_tools/templates/create/lib/main.dart.tmpl) that's created with each new Flutter project.
```
Comment (abarth) [inline index.md]: Please link to `master` instead of `fa2caa737bdde4d72525d9d4260abb0ba5548c01`

File: index.md [hunk 30]
```diff
@@ -1,63 +1,200 @@
...
+
+Having separate state and widget objects lets other widgets treat stateless and 
+stateful widgets in the same way, without being concerned about losing state. 
+Rather than needing to hold on to the child to preserve its state, the parent is
```
Comment (abarth) [inline index.md]: to a child

**Comment by abarth** [comment]
This is looking pretty good.  A bunch of minor comments above.  Also, your patch has a bunch of changes to other pages on the site that look like bad merges.  We'll need to clean those up before we can land your patch.

File: index.md (hunk 2 above)
Comment (LarkAscending) [inline index.md]: We have that great domain name - one that people can remember. Search engines also very much prefer when the domain name matches the product name. All good there. Also, we'll deal with it shortly with metadata in _config.yml, courtesy of an SEO gem, instead of actual text. Can put whatever your heart desires in there, without consequence.

File: _includes/layer-cake.html (hunk 16 above)
Comment (LarkAscending) [inline _includes/layer-cake.html]: Thanks for looking! Because the people I talked to about Custom (not you) thought these were customized versions of material widgets.

What would you point it to?

File: index.md (hunk 17 above)
Comment (LarkAscending) [inline index.md]: Agreed. This is from @sethladd and I'll trust he'll respond if he doesn't like the change.

File: index.md (hunk 2 above)
Comment (sethladd) [inline index.md]: That's true, but people don't search for "flutter", they search for "mobile app sdk" or "mobile app framework", etc.

Here's what we look like today: it's descriptive and helpful:

screen shot 2017-03-01 at 10 08 28 am

File: index.md (hunk 17 above)
Comment (sethladd) [inline index.md]: hah good catch. There's probably a better way to do this. We wanted to point out:

* our attention to detail to the scrolling frictions and feels
* overscroll behaviors are different.

Maybe we could s/scrolling behaviors/the feel of scrolling ?

File: index.md (hunk 18 above)
Comment (LarkAscending) [inline index.md]: The grammar is totally legit and the reason the series groups together buttons and dropdown lists is they are both traditional UI controls. The list used to say "controls like..." We can say "along with a shopping cart page" to differentiate it, if that would help.

On the phone comment, this wording is meant to be a relief from the cluster of "iOS and Android" references above. It's a trigger, a nudge to just go get the Gallery and run it on a phone. Don't even have to set up an emulator, or an IDE, etc. Anyway, does it work for that purpose?

File: index.md (hunk 21 above)
Comment (LarkAscending) [inline index.md]: Yes, it was that way, if you check one commit back. There's a controversy! @sethladd can you help us all get on the same page?

Comment (sethladd) [inline index.md]: Yeah, I think it was lower-case previously (I recall other reviews where we were asked to lower case the d in design). Happy to defer to the current material.io website.

File: index.md (hunk 22 above)
Comment (LarkAscending) [inline index.md]: The line break does not matter - it does not show.

Changed The >These

File: index.md (hunk 19 above)
Comment (LarkAscending) [inline index.md]: Thanks!

File: index.md (hunk 20 above)
Comment (LarkAscending) [inline index.md]: Done. (not sure why, but will research it)

File: index.md (hunk 23 above)
Comment (LarkAscending) [inline index.md]: Done.

File: index.md (hunk 24 above)
Comment (LarkAscending) [inline index.md]: True - it doesn't show b/c Markdown is smart, but removed it so no one trips over it again.

File: index.md (hunk 25 above)
Comment (LarkAscending) [inline index.md]: Sorry, lost my nerve there and made it boring. Changed back to delicious! (Could also change to "good for you"...)

File: index.md (hunk 26 above)
Comment (LarkAscending) [inline index.md]: Fixed.

File: index.md (hunk 27 above)
Comment (LarkAscending) [inline index.md]: Done

File: index.md (hunk 28 above)
Comment (LarkAscending) [inline index.md]: OK (usually no possessive, but this case is different)

File: index.md (hunk 30 above)
Comment (LarkAscending) [inline index.md]: Done

File: index.md (hunk 21 above)
Comment (LarkAscending) [inline index.md]: There's a brand new home page where they spell it out! Material...Design!!! So exciting.
Glad it's clear now... before it was (kinda) lower case "d" but they weren't being explicit.

File: index.md (hunk 29 above)
Comment (LarkAscending) [inline index.md]: Roger that.

File: android-release-build.md [hunk 31]
```diff
@@ -0,0 +1,159 @@
+---
+layout: page
+title: Preparing an Android App for Release
```
Comment (abarth) [inline android-release-build.md]: This looks like a stray diff

File: building-apk.md [hunk 32]
```diff
@@ -1,33 +0,0 @@
----
-layout: page
```
Comment (abarth) [inline building-apk.md]: Another stray diff

**Comment by sethladd** [comment]
I think we merged this in from another PR?

**Comment by sethladd** [comment]
Closing because I think this was replaced by another PR. Please reopen if we still wanted to merge this in.
